components/form/ProjectForm.tsx

```diff
@@ -0,0 +1,195 @@
+import { Button, FormControl, FormLabel, Input, Select, FormErrorMessage } from '@chakra-ui/react';
+import { Flex } from '@chakra-ui/layout';
+import React, { useState } from 'react';
+import { useRouter } from 'next/router';
+import ErrorAlert from '@/components/common/ErrorAlert';
+import { ProjectDTO, EmployeeDTO, CustomerDTO, ProjectStatus } from '@/lib/types/dto';
+import CustomerForm from './CustomerForm';
+import { postProject, putProject } from '@/lib/utils/apiRequests';
+
+type ProjectFields = Partial<ProjectDTO> & { status: ProjectStatus };
+
+const validateProjectFields = (form: ProjectFields): ProjectDTO => {
+    const { name, startDate, customer, managingEmployee, status } = form;
+    if (name && startDate && customer && managingEmployee && status) {
+        return {
+            ...form,
+            name,
+            startDate,
+            customer,
+            managingEmployee,
+            status,
+        };
+    } else {
+        throw 'Invalid project form: missing required fields';
+    }
+};
+
+type ProjectFormPropsBase = {
+    employees: EmployeeDTO[];
+    customers: CustomerDTO[];
+    refreshCustomers: () => void;
+};
+
+type ProjectFormProps =
```

Comment:
The type `ProjectFormProps` should reflect the fact that `project` is defined iff `method=put` and is `undefined` otherwise.